src/features/courses/course-repository.ts

```diff
@@ -573,6 +573,17 @@ class CourseRepository {
         }
     }
     
+    async getStudentGradeInstance(options: GetStudentGradeInstanceOptions): Promise<StudentGradeInstance> {
+        const result = await StudentGradeInstance.findOne({
+            where: {
+                id: options.id,
+                active:true
+            }
+        });
+        if (_.isNil(result)) throw new NotFoundError(`Requested grade instance #${options.id} not found`);
```

Comment:
I'm back and forth on this, I feel like it should be the caller's responsibility to null check it
I'm fine either way though